### DISCUSSION TITLE
What is the event emitted when i press on the backdrop of the q-drawer ?

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#10749] [General - Components / Directives / etc]
Hi, i have a layout that contains a drawer, now i am opening and closing the drawer through mutations in vuex. I want to know what is the event emitted by clicking on the backdrop when the drawer is open so i can use it and change the `isLeftDrawerOpen` state instead of following the approach below.

`src/store/MainLayout/state.js`

```js
export default function () {
  return {
    drawer: {
      isLeftDrawerOpen: false
    }
  }
}
```

`src/store/MainLayout/mutations.js`
```js
export function toggleLeftDrawer(state) {
  state.drawer.isLeftDrawerOpen = !state.drawer.isLeftDrawerOpen
}
export function hideLeftDrawer(state) {
  state.drawer.isLeftDrawerOpen = false
}
```

`src/layouts/MainLayout.vue`

```vue
<template>
  <q-layout view="hhh lpR lff">
    <q-header reveal elevated class="bg-primary text-white">
      <q-toolbar>
        <q-btn dense flat round icon="menu" @click="toggleLeftDrawer" />

        <q-toolbar-title>
          <q-avatar>
            <img src="https://cdn.quasar.dev/logo-v2/svg/logo-mono-white.svg" />
          </q-avatar>
          Title
        </q-toolbar-title>
      </q-toolbar>
    </q-header>

    <Drawer />

    <q-page-container>
      <router-view />
    </q-page-container>
  </q-layout>
</template>

<script>
import { ref, defineAsyncComponent } from "vue";
import { useStore } from "vuex";

export default {
  components: {
    Drawer: defineAsyncComponent(() => import("components/UI/Drawer.vue")),
  },
  setup() {
    const $store = useStore();

    const toggleLeftDrawer = () => {
      $store.commit("MainLayout/toggleLeftDrawer");
    };

    return {
      toggleLeftDrawer,
    };
  },
};
</script>
```

`src/components/UI/Drawer.vue`

```vue
<template>
  <q-drawer
    :model-value="data.isLeftDrawerOpen"
    side="left"
    overlay
    elevated
    :width="250"
    @hide="onDrawerHidden"
  >
    drawer content
  </q-drawer>
</template>

<script>
import { ref, computed } from "vue";
import { useStore } from "vuex";
export default {
  setup() {
    const $store = useStore();

    const data = computed(() => {
      return $store.state.MainLayout.drawer;
    });

    const onDrawerHidden = () => {
      $store.commit("MainLayout/hideLeftDrawer");
    };

    return {
      data,
      onDrawerHidden,
    };
  },
};
</script>

```

**Comment by MilosPaunovic** [comment]
Have you tried inspecting events on your page using VueDevtools?

**Reply by ghost** [reply]
Thank you for reply, i will try this and comment the results.

**Reply by ghost** [reply]
The Devtools were not useful, I did not find any events in the VueDevtools.

**Reply by MilosPaunovic** [reply]
Have you looked in `events` tab?

**Reply by ghost** [reply]
> Have you looked in `events` tab?

I don't find this events tab

**Reply by MilosPaunovic** [reply]
![Screenshot 2021-09-21 at 10 28 50](https://user-images.githubusercontent.com/62475782/134138137-5f1ba16d-309f-46c8-bc1c-678e55f846eb.png)

**Reply by ghost** [reply]
![113034](https://user-images.githubusercontent.com/76194919/134138484-cc9a150b-c5d6-4cd2-a740-d24afe5a217e.png)

**Comment by ontwikkelfabriek** [comment]
@FuridamuFaita 

>  I want to know what is the event emitted

To find out , you can take a look at the source code of Quasar q-drawer:
https://github.com/quasarframework/quasar/blob/dev/ui/src/components/drawer/QDrawer.js

btw that doesn't mean it is accessible....

**Reply by ghost** [reply]
I already looked and i didn't find anything.

**Reply by ontwikkelfabriek** [reply]
@FuridamuFaita 

This could be the one:
https://github.com/quasarframework/quasar/blob/dev/ui/src/composables/private/use-model-toggle.js

Even if you can listen to the undocumented event( used internally by Quasar). This is probably not a good idea. I would always prefer a solution that uses the public API of a Component. In other words , I would not change your current solution.

**Reply by ghost** [reply]
Thanks for your help

**Comment by cyfung1031** [comment]
For v2, almost all the things are private in QDrawer, there are no event triggering. This is the thing I dislike most for Quasar.
Similar Question - not able to control the pulling of drawer #10746

The development team shall consider to give more events and methods for QDrawer.

**Comment by cyfung1031** [comment]
I have just made a PR for this feature. #10761
